test/tbb/test_task.cpp

```diff
@@ -794,3 +794,16 @@ TEST_CASE("Test with priority inversion") {
     }
     obsr.observe(false);
 }
+
+// Explicit test for raii_guard move ctor because of copy elision optimization
```

Comment:
```suggestion
// Explicit test for raii_guard move ctor because of copy elision optimization
// TODO: consider better test file for the test case
```